frontend/src/components/TraineesAutocomplete/TraineesAutocomplete.tsx

```diff
@@ -0,0 +1,58 @@
+import useSWR from 'swr';
+import { TraineeFindAllResponseDto } from '@osk/shared';
+import { useState } from 'react';
+import { addQueryParams } from '../../utils/addQueryParams';
+import { GeneralAPIError } from '../GeneralAPIError/GeneralAPIError';
+import { FAutocomplete } from '../FAutocomplete/FAutocomplete';
+import { useDebounce } from '../../hooks/useDebounce/useDebounce';
+
+interface TraineesAutocompleteProps {
+  required: boolean;
+}
+
+export const TraineesAutocomplete = ({
+  required,
+}: TraineesAutocompleteProps) => {
+  const [inputValue, setInputValue] = useState<string>('');
+  const [savedInputValue, setSavedInputValue] = useState<string | null>(null);
+  const debouncedValue = useDebounce(inputValue);
+  const { data, error, isValidating } = useSWR<TraineeFindAllResponseDto>(
+    debouncedValue
+      ? addQueryParams('/api/trainees', {
```

Comment:
Let's specify the type of the filters

```suggestion
      ? addQueryParams<TraineeFindAllQueryDto>('/api/trainees', {
```